File: Java/0101-0150/104. Maximum Depth of Binary Tree.java
// https://leetcode.com/problems/maximum-depth-of-binary-tree/description/

class Solution {
    int maxDepth = 0;
    public int maxDepth(TreeNode root) {
        if(root == null){
            return 0;
        }

        //first left and then right
        int left = maxDepth(root.left);
        int right = maxDepth(root.right);

        int height = left + right + 1;

        maxDepth = Math.max(maxDepth, height);

        return Math.max(left,right)+1;
    }
}

File: Javascript/0101-0150/104. Maximum Depth of Binary Tree.js
// https://leetcode.com/problems/maximum-depth-of-binary-tree/description/

 let Depth = 0;
var maxDepth = function(root) {
    if(root == null){
        return 0;
    }

    //first left and then right
    const left = maxDepth(root.left);
    const right = maxDepth(root.right);

    let height = left + right + 1;

    Depth = Math.max(Depth, height);

    return Math.max(left,right)+1;
};
